astroplan/tests/test_core.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from astropy.coordinates import (EarthLocation, Latitude, Longitude, SkyCoord)
+import astropy.units as u
+from astropy.time import Time
+from astropy.tests.helper import remote_data
+import numpy as np
+import ephem
+import pytz
+
+from ..core import FixedTarget, Observer
+
+
+def test_Observer_constructor_location():
+    '''
+    Show that location defined by latitude/longitude/elevation is parsed
+    identically to passing in an `~astropy.coordinates.EarthLocation` directly.
+    '''
+
+    lat = '+19:00:00'
+    lon = '-155:00:00'
+    elevation = 0.0 * u.m
+    location = EarthLocation.from_geodetic(lon, lat, elevation)
+
+    environment_kwargs = dict(pressure=1*u.bar, relative_humidity=0.1,
+                              temperature=10*u.deg_C)
+
+    obs1 = Observer(name='Observatory',
+                    latitude=lat,
+                    longitude=lon,
+                    elevation=elevation,
+                    **environment_kwargs)
+
+    obs2 = Observer(name='Observatory',
+                    location=location,
+                    **environment_kwargs)
+
+    assert obs1.location == obs2.location, ('Locations not parsed equivalently')
```

Comment:
@eteq – I agree leaving a note on the purpose of a given test / assert can be useful.
So far I have been doing this in docstrings or comments in tests, usually not at the place of the assert, but higher up where I start a new test. As far as I can see that's also the common thing to do in Astropy:
https://github.com/astropy/astropy/search?p=1&q=assert&utf8=%E2%9C%93

Another disadvantage if putting this note after the assert is that often you're almost at the end of the line, and then you have to wrap a lot (see the examples below in this PR).

But my main point is that asserts, for written and debugged code will very rarely fail, and thus it's just not worth to implement extra notes for every assert. If someone has a test failure, it's always worth reporting.
The user should copy & paste the console output into the issue report or a gist, and then a developer / maintainer will look at the test code anyways.

But bottom line: this is probably partly a matter of taste, and @bmorris3, if you like it this way, we'll still merge this tomorrow latest. :-)